spec-examples/src/aes_gcm/aesgcm.rs

```diff
@@ -36,86 +35,84 @@ fn pad_aad_msg(aad: &ByteSeq, msg: &ByteSeq) -> ByteSeq {
     padded_msg
 }
 
-pub fn encrypt_aes128(
-    key: aes::Key128,
+pub(crate) fn encrypt_aes(
+    key: &ByteSeq,
     iv: aes::Nonce,
     aad: &ByteSeq,
     msg: &ByteSeq,
+    alg: aes::AesVariant,
 ) -> (ByteSeq, Tag) {
     let iv0 = aes::Nonce::new();
 
-    let mac_key = aes128_ctr_keyblock(key, iv0, U32(0), 4, 10);
-    let tag_mix = aes128_ctr_keyblock(key, iv, U32(1), 4, 10);
+    let mac_key = aes_ctr_keyblock(key, iv0, U32(0), aes::nk(alg), aes::nr(alg), alg);
+    let tag_mix = aes_ctr_keyblock(key, iv, U32(1), aes::nk(alg), aes::nr(alg), alg);
 
-    let cipher_text = aes128_encrypt(key, iv, U32(2), msg);
+    let cipher_text = aes_encrypt(key, iv, U32(2), msg, alg);
     let padded_msg = pad_aad_msg(aad, &cipher_text);
     let tag = gmac(&padded_msg, Key::from_seq(&mac_key));
     let tag = aes::xor_block(Block::from_seq(&tag), tag_mix);
 
     (cipher_text, Tag::from_seq(&tag))
 }
 
+pub fn encrypt_aes128(
+    key: aes::Key128,
+    iv: aes::Nonce,
+    aad: &ByteSeq,
+    msg: &ByteSeq,
+) -> (ByteSeq, Tag) {
+    encrypt_aes(&ByteSeq::from_seq(&key), iv, aad, msg, aes::AesVariant::Aes128)
+}
+
 pub fn encrypt_aes256(
     key: aes::Key256,
     iv: aes::Nonce,
     aad: &ByteSeq,
     msg: &ByteSeq,
 ) -> (ByteSeq, Tag) {
-    let iv0 = aes::Nonce::new();
-
-    let mac_key = aes256_ctr_keyblock(key, iv0, U32(0), 8, 14);
-    let tag_mix = aes256_ctr_keyblock(key, iv, U32(1), 8, 14);
-
-    let cipher_text = aes256_encrypt(key, iv, U32(2), msg);
-    let padded_msg = pad_aad_msg(aad, &cipher_text);
-    let tag = gmac(&padded_msg, Key::from_seq(&mac_key));
-    let tag = aes::xor_block(Block::from_seq(&tag), tag_mix);
-
-    (cipher_text, Tag::from_seq(&tag))
+    encrypt_aes(&ByteSeq::from_seq(&key), iv, aad, msg, aes::AesVariant::Aes256)
 }
 
-pub fn decrypt_aes128(
-    key: aes::Key128,
+pub(crate) fn decrypt_aes(
```

Comment:
Does really need to be `pub`?

Reply:
At least `pub(crate)` because `aes_gcm` is also implemented generically, so it has to be able to call the generic functions